Wangscape/noise/Reseedable.h

```diff
@@ -43,6 +44,46 @@ struct Reseedable final
     Reseedable scalePoint(double x_scale, double y_scale, double z_scale);
     Reseedable translatePoint(double x_displace, double y_displace, double z_displace);
     Reseedable turbulence(double frequency, double power, int roughness, int seed);
+    template<typename Iterator>
+    Reseedable terrace(Iterator begin, Iterator end, bool inverted = false);
+    Reseedable terrace(int controlPointCount, bool inverted = false);
+    template<typename Iterator>
+    Reseedable curve(Iterator begin, Iterator end);
+private:
+    Reseedable finaliseTerrace(std::shared_ptr<module::Terrace>& terrace_ptr);
+    Reseedable finaliseCurve(std::shared_ptr<module::Curve>& curve_ptr);
```

Comment:
Passing it by reference skips the reference count increment/decrement/comparison. But this part of the code isn't performance-sensitive (and it might even be optimised out), so it's probably not worthwhile.